ptsites/schema/gazelle.py

```diff
@@ -54,7 +55,8 @@ def details_selector(self) -> dict:
             }
         }
 
-    def get_gazelle_message(self, entry, config, message_body_selector='div[id*="message"]'):
+    def get_gazelle_message(self, entry: SignInEntry, config: dict,
+                            message_body_selector: str = 'div[id*="message"]') -> None:
```

Comment:
无意义的类型定义 使用类型推断

Reply:
@madwind 你指的是由于 `messge_body_selector: str` 的默认参数为 `str` 所以不需要显示添加 `str` 吗？

这是错误的。当没有指定函数的参数类型时，类型检查器会将参数推断为 `Any`，[PEP 484](https://peps.python.org/pep-0484/) 没有将拥有默认参数的情况设为例外。

mypy(Python官方的类型检查器)不会对以下代码进行报错：
```python
def f(a=''):
    print(a)

f(1)
```
PyCharm会报错，但这并不是标准。
还有不懂的请参考 https://github.com/python/mypy/issues/3090

Reply:
你可以看一下Mypy核心开发者写的：

Mypy follows PEP 484. A function parameter without an annotation is the same as having an `Any` annotation, and there is no exception for default values. Here is some rationale for this:

* Often the default argument is not enough to infer a type. For example, `None` doesn't give enough context. With the current rules this doesn't pose a difficulty.
* The default argument may not be enough to infer the *correct* type. For example, if the default is `''` in Python 2, the correct type could well be `Union[str, unicode]`. If the default is `0`, the correct type might well be `float`. If the programmer understands the rules for type inference they could override the default type with an annotation as needed, but this would add some extra complexity.
* The default value could be a complex expression such as a function call, and thus the type may not obvious from just the default value. Having an annotation makes code like this easier to read.
* If a function has no annotation at all, then the argument type would have to be `Any` anyway. The current rule makes this consistent across functions with no annotations and functions with partial annotations.

It all boils down to the current rule being simple and obvious, and in practice it doesn't feel much of a burden to add a few extra `: int` annotations. There is no deep technical reason for the current rule, though it makes type inference easier for mypy.

_Originally posted by @JukkaL in https://github.com/python/mypy/issues/3090#issuecomment-290691282_